### PR TITLE
Boot Grafana and Prometheus in docker-compose

### PR DESCRIPTION
**Pull request by lawrencejones**
This commit adds Grafana and Prometheus containers to the docker-compose
stack, enabling developers to locally verify their metrics, alerts and
dashboards. Some of the alerts are a bit suspect right now but this is
the first step in making them good.

Also ensure Circle doesn't boot these containers when running acceptance
tests. While it doesn't cause the test to fail, there's no point
starting prometheus and friends in an integration testing environment.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -39,9 +39,9 @@ services:
     entrypoint:
       - /stolon-pgbouncer/bin/stolon-pgbouncer.linux_amd64
       - supervise
+      - --metrics-address=0.0.0.0
       - --enable-child-process
       - --pgbouncer-config-template-file=/stolon-pgbouncer/docker/stolon-development/pgbouncer/pgbouncer.ini.template
-      - --poll-interval=5s
```
Comment (hmac) [inline docker-compose.yml]: ?

File: docker/observability/prometheus/rules.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            $ kubectl -n {{ $labels.namespace }} exec -it {{ $labels.pod_name }} bash
+            postgres:/$ ps auxf
+            ... # expect to see PgBouncer and stolon-pgbouncer
+            postgres:/$ psql -p 6432 -U pgbouncer pgbouncer -h /tmp
+            psql (11.2 (Ubuntu 11.2-1.pgdg18.04+1), server 1.9.0/bouncer)
+            Type "help" for help.
+
+            pgbouncer=# show databases;
+            ...
+            ```
```
Comment (hmac) [inline docker/observability/prometheus/rules.yml]: Is this duplicated with rules that we have internally at GC?

File: docker-compose.yml (hunk 1 above)
Comment (lawrencejones) [reply docker-compose.yml]: Ah this was removed as it doesn't help our tests and makes the dashboard a bit strange as it assumes a minute value.

To be clearer, stolon-pgbouncer uses an etcd watch so should update immediately, which is why reducing this value wouldn't improve our CI times.

File: docker/observability/prometheus/rules.yml (hunk 2 above)
Comment (lawrencejones) [reply docker/observability/prometheus/rules.yml]: It is slightly yeah, minus some things like our dashboard link. Not sure what the best way to go around this is though, definitely worth having them in this repo for local testing but quite difficult to sync up.

**Pull request merged**